tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/symbols/TypeSymbolTest.java

```diff
@@ -78,48 +80,51 @@ public void setup() {
     }
 
     @Test(dataProvider = "PosProvider")
-    public void testTypeSymbolLookup(int line, int col, TypeDescKind typeKind, String signature) {
+    public void testTypeSymbolLookup(int line, int col, TypeDescKind typeKind, String signature, LineRange lineRange) {
         TypeSymbol type = assertBasicsAndGetType(line, col, typeKind, signature);
         assertTrue(type.getName().isEmpty());
-        assertTrue(type.getLocation().isEmpty());
+        assertEquals(type.getLocation().map(Location::lineRange).orElse(null), lineRange);
     }
 
     @DataProvider(name = "PosProvider")
     public Object[][] getTypePosition() {
         return new Object[][]{
-                {17, 4, INT, "int"},
-                {18, 4, BOOLEAN, "boolean"},
-                {19, 4, FLOAT, "float"},
-                {20, 4, DECIMAL, "decimal"},
-                {21, 4, NIL, "()"},
-                {22, 4, STRING, "string"},
-                {23, 4, INT, "int"},
+                {17, 4, INT, "int", null},
+                {18, 4, BOOLEAN, "boolean", null},
+                {19, 4, FLOAT, "float", null},
+                {20, 4, DECIMAL, "decimal", null},
+                {21, 4, NIL, "()", null},
+                {22, 4, STRING, "string", null},
+                {23, 4, INT, "int", null},
 //                {26, 4, TypeDescKind.XML, "xml"},
 //                {27, 4, TypeDescKind.XML, "xml<xml>"},
 //                {28, 4, TypeDescKind.XML, "xml<ballerina/lang.xml:0.8.0:Element>"},
-                {30, 4, MAP, "map<string>"},
-                {31, 4, TYPEDESC, "typedesc<anydata>"},
-                {32, 4, TYPEDESC, "typedesc<any|error>"},
-                {33, 4, TABLE, "table<Person>"},
-                {34, 4, TABLE, "table<Person> key<int>"},
-                {35, 4, FUTURE, "future<string>"},
-                {36, 4, FUTURE, "future<any|error>"},
-                {40, 4, INT, "int"},
-                {46, 4, ERROR, "error<ErrorData>"},
-                {47, 4, HANDLE, "handle"},
-                {48, 4, STREAM, "stream<Person, error>"},
-                {52, 4, SINGLETON, "10"},
-                {54, 4, ANY, "any"},
-                {55, 4, NEVER, "never"},
-                {56, 4, READONLY, "readonly"},
-                {57, 4, INT, "int"},
-                {57, 8, STRING, "string"},
-                {58, 13, READONLY, "readonly"},
-                {59, 4, INT, "int"},
-                {60, 4, ANYDATA, "anydata"},
-                {61, 4, JSON, "json"},
-                {62, 4, BYTE, "byte"},
-                {63, 13, ERROR, "error"},
+                {30, 4, MAP, "map<string>",
+                        LineRange.from(srcFile.name(), LinePosition.from(30, 4), LinePosition.from(30, 15))},
```

Comment:
instead of having `LineRange.from(srcFile.name()...` shall we have only `line` and `col` values. Then, let's check the **if** condition I mentioned previously appropriately

Reply:
I thought of this initially, but this would result in polluted null keywords for each such parameter.

Reply:
But with this way, you have to use `LineRange.from(srcFile.name()..` more frequently right?